src/main/java/entities/GameManager.java

```diff
@@ -0,0 +1,231 @@
+package entities;
+
+import java.util.ArrayList;
+
+/**
+ * Proxy class for Game.
+ * 
+ * Allows interactors to reference a shared game without one being made.
+ */
+public class GameManager implements CreationAccess,
+                                    GameAccess,
+                                    ObservableGame {
+    private Game game;
+    private ArrayList<GameObserver> observers;
+
+    /**
+     * Constructs a GameManager with empty attributes.
+     */
+    public GameManager() {
+        this.game = null;
+        this.observers = new ArrayList<>();
+    }
+
+    /**
+     * Updates observers of this game.
+     */
+    @Override
+    public void notifyGameObservers() {
+        for (GameObserver observer : observers) {
+            observer.updateGameObserver(this);
+        }
+    }
+
+    /**
+     * Adds an observer to this game.
+     */
+    @Override
+    public void addObserver(GameObserver observer) {
+        observers.add(observer);
+    }
+
+    /**
+     * Removes an observer from this game.
+     */
+    @Override
+    public void deleteObserver(GameObserver observer) {
+        observers.remove(observer);
+    }
+
+    /**
+     * Removes all observers from this game.
+     */
+    @Override
+    public void deleteObservers() {
+        observers.clear();
+    }
+
+    /**
+     * Gets the player whose turn it currently is.
+     * @return the current player
+     */
+    @Override
+    public Player getCurrentTurn() {
```

Comment:
All these methods are already tested, but I would suggest a few tests just checking for null.